source/extensions/filters/network/thrift_proxy/transport.h

```diff
@@ -0,0 +1,163 @@
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "envoy/buffer/buffer.h"
+
+#include "common/singleton/const_singleton.h"
+
+#include "absl/types/optional.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace ThriftProxy {
+
+/**
+ * Names of available Transport implementations.
+ */
+class TransportNameValues {
```

Comment:
I have the same suggestion for this as for `ProtocolNameValues`: make each subclass of `Transport` define its own name string, rather than collecting the names in a centralized class that has to be updated every time there's a new `Transport` subclass.